### PR TITLE
Enhancement/no map in void context

### PR DESCRIPTION
**Pull request by mcmillhj**
patch for issue#30, remove map statements that don't produce a resulting list in favor of foreach loops

### CODE REVIEW

**Comment by mcmillhj** [comment]
That is weird, all tests for this pass in my local branch.

File: lib/Interchange6/Cart.pm [hunk 1]
```diff
@@ -201,12 +202,13 @@ has weight => (
...
 sub _build_weight {
     my $self = shift;
-    my $weight;
-
-    map { $weight += $_->weight * $_->quantity }
-      grep { defined $_->weight } $self->products_array;
+    
+    my $weight = 0;
+    foreach my $product ( grep { defined $_->weight } $self->products_array ) {
+        $weight += $product->weight;
```
Comment (SysPete) [inline lib/Interchange6/Cart.pm]: Looks like you dropped quantity - this works:
`$weight += $product->weight * $product->quantity;`

**Comment by SysPete** [comment]
@mcmillhj Any chance you could remove the nasty merge commit of master and just rebase once you've updated the code? Don't worry if that is a problem as I can always cherry-pick commits from your branch.

Cheers!

**Comment by mcmillhj** [comment]
Sure, to do that I just need to:

```
git checkout <branch>
git rebase upstream/master
```
right?

**Comment by mcmillhj** [comment]
Hmm, not sure what I have done here. I can always delete this branch and re-created since the changes are minimal.

**Comment by SysPete** [comment]
Looking at your current `enhancement/no-map-in-void-context` branch I'd be doing something along the lines of...
```
git checkout master
git pull upstream master # if you have problems here then merge --abort and reset --hard a few commits and try again
git checkout enhancement/no-map-in-void-context
git reset --hard bbf8f065efa213552d553d6f805d3c01ee584c8d # junk the merge commit
git rebase master
```

**Comment by SysPete** [comment]
And I see travis is catching you out with the new test file missing from MANIFEST. Happens to me all the time too which is why we run RELEASE_TESTING there. Just wish I could always remember to run release tests before I push. :smile:

**Comment by mcmillhj** [comment]
Hmm, I just realized that there is another branch merged into these changes that is actually for a different issue. Sorry, not super used to submitting multiple PRs at a time for the same repo. I need to revert the merge of https://github.com/interchange/Interchange6/commit/bf4ae5296155ef8f068c59fbfcdebbf461119876

**Comment by SysPete** [comment]
No worries. I understand completely as I spend most of my time pushing directly to master in most projects. Only when I'm working on something like Dancer2 do I need to remember to branch and keep my PRs clean and separated. Messed up so many times when I first started working like that.

**Comment by mcmillhj** [comment]
Starting fresh.

**Pull request closed without merging**